packages/components/src/link/index.js

```diff
@@ -3,40 +3,47 @@
  * External dependencies
  */
 import { Component } from '@wordpress/element';
-import { Link as RouterLink } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
 /**
  * WooCommerce dependencies
  */
-import { getAdminLink } from '@woocommerce/navigation';
+import { getAdminLink, getHistory } from '@woocommerce/navigation';
 
 /**
  * Use `Link` to create a link to another resource. It accepts a type to automatically
  * create wp-admin links, wc-admin links, and external links.
  */
 class Link extends Component {
+	// @todo Investigate further if we can use <Link /> directly.
+	// With React Router 5+, <RouterLink /> cannot be used outside of the main <Router /> elements,
+	// which seems to include components imported from @woocommerce/components. For now, we can use the history object directly.
```

Comment:
I agree, there is some potential for simplification. We could remove `getNewPath` which has potential consequences (accepts two query objects, one optional that defaults to the current query), in favor of a more direct approach that simply accepts an object. I think best to follow up on this afterwards.